server/npg/porch/endpoints/tasks.py

```diff
@@ -27,133 +28,159 @@
-        status.HTTP_404_NOT_FOUND: {
-            "description": "The pipeline or task in the request is invalid"
-        },
-    }
+    },
+    summary="Update one task.",
+    description='''
+    Given a Task object, updates the status of the task in the database
+    to the value of the status in this Task object.
+
+    If the task does not exist, status 404 'Not found' is returned.'''
 )
-async def update_task(task: Task, db_accessor=Depends(get_DbAccessor)):
-    """
-    Given a Task object, updates the status of the task in the database.
+async def update_task(
+    task: Task,
+    db_accessor=Depends(get_DbAccessor),
+    permission=Depends(validate)
+) -> Task:
 
-    The pipeline specified by the `pipeline` attribute of the Task object
-    should exist. If it does not exist, return status 404 'Not found' and
-    an error.
-    """
+    _validate_request(permission, task.pipeline)
     changed_task = None
     try:
-        changed_task = await db_accessor.update_task(token_id=1, task=task)
+        changed_task = await db_accessor.update_task(
+            token_id=permission.requestor_id,
+            task=task
+        )
     except NoResultFound as e:
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=str(e))
+
     return changed_task
 
+
 @router.post(
     "/claim",
     response_model=List[Task],
-    summary="Claim tasks.",
-    description="Claim tasks for a particular pipeline.",
     responses={
-        status.HTTP_200_OK: {"description": "Receive a list of tasks that have been claimed"},
-        status.HTTP_404_NOT_FOUND: {
-            "description": "Cannot find the pipeline submitted with the claim"
-        }
-    }
-)
-async def claim_task(
-    pipeline: Pipeline,
-    num_tasks: PositiveInt = 1,
-    db_accessor=Depends(get_DbAccessor)
-) -> List[Task]:
-    """
+        status.HTTP_200_OK: {"description": "Receive a list of tasks that have been claimed"}
+    },
+    summary="Claim tasks for a particular pipeline.",
+    description='''
     Arguments - the Pipeline object and the maximum number of tasks
     to retrieve and claim, the latter defaults to 1 if not given.
 
-    Return an error and status 404 'Not Found' if the pipeline with the
-    given name does not exist.
+    If no tasks that satisfy the given criteria and are unclaimed
+    are found, returns status 200 and an empty array.
 
-    It is possible that no tasks that satisfy the given criteria and
-    are unclaimed are found. Return status 200 and an empty array.
+    If any tasks are claimed, return an array of these Task objects
+    and status 200.
 
-    If any tasks are claimed, return an array of these Task objects and
-    status 200.
+    The pipeline object returned within each of the tasks is consistent
+    with the pipeline object in the payload, but has all possible
+    attributes defined (uri, version).'''
+)
+async def claim_task(
+    pipeline: Pipeline,
```

Comment:
pipeline is redundant with permission now, keeping for verifying what the client meant?

Reply:
Firstly, we need to know the pipeline to validate the token, secondly, it does not harm to have this stated explicitly and, finally, the auth scheme might change.